docs: Update macOS development instructions

mastion opened the pull request:
Updating macOS development readme with some gotchas that I ran into while getting setup.
 - Linked to collab readme because that contained the steps to setup the postgres database so integration tests pass
 - Added section under troubleshooting. Recommending `cargo-nextest` since the CI uses it and it got me past the failures I was seeing.

Release Notes:

- N/A

mastion committed:
first pass at some issues we ran into

mastion committed:
Adjusting verbiage on doc

Co-Authored-By: KyleBarton <kjbarton4@gmail.com>

mastion committed:
Fixing spelling mistake

Co-Authored-By: KyleBarton <kjbarton4@gmail.com>

mastion committed:
removes extra whitespace

Co-Authored-By: KyleBarton <kjbarton4@gmail.com>

maxdeviant commented on docs/src/development/macos.md:
````diff
@@ -41,6 +41,7 @@ If you are developing collaborative features of Zed, you'll need to install the
   ```sh
   brew install livekit foreman
   ```
+- Follow the steps in the [collab README](../../../crates/collab/README.md) to configure the Postgres database for integration tests
````
We can't use relative links like this in the docs, as they will break when deployed to zed.dev/docs.

This needs to be an absolute URL to GitHub.

mastion replied:
Should be fixed

mastion committed:
use an github url to link to the collab readme

mastion committed:
fixed with prettier

Command from CI: cd docs && pnpm dlx prettier@${PRETTIER_VERSION} . --write && cd ..

Co-Authored-By: KyleBarton <kjbarton4@gmail.com>

SomeoneToIgnore reviewed (approved):
Sorry for the extra noise in the commits, have been trying a few things with the relative URLs.

Thank you for the adjustments.
